lib/core/filtering/post_process_engine.rb

```diff
@@ -0,0 +1,38 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+# frozen_string_literal: true
+
+require 'core/filtering'
+
+module Core
+  module Filtering
+    class PostProcessEngine
+      attr_reader :rules
+
+      def initialize(job_description)
+        @rules = ordered_rules(job_description['filtering'])
```

Comment:
There's now a utility function in main that does this extraction (in `Utility::Filtering`). I think we should use that. Or alternatively, create a wrapper around the job entity with the methods we need, like it's been done with connectors config.